waltz-ng/client/data-flow/data-flow-edit.js

```diff
@@ -202,11 +202,20 @@ function controller($q,
     };
 
     vm.addSource = (app) => {
+        if(vm.app.id == app.id) {
+            notification.warning("An application may not link to itself.");
```

Comment:
Good idea. I'll put it in.

On 3 Aug 2016 18:47, "David Watkins" <notifications@github.com> wrote:

> In waltz-ng/client/data-flow/data-flow-edit.js
> <https://github.com/khartec/waltz/pull/378#discussion_r73377842>:
>
> > @@ -202,11 +202,20 @@ function controller($q,
> >      };
> >
> >      vm.addSource = (app) => {
> > +        if(vm.app.id == app.id) {
> > +            notification.warning("An application may not link to itself.");
>
> should this be pulled out as it's common
>
> If (notifyIllegalFlow(vm.app,app)) return;
>
> Or similar ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/378/files/b4416f69dd1d4ee54eabb3fef35fa0d0a9756043#r73377842>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC45272W8xISiqT0ijwzio6Fvt25yrfXks5qcNMCgaJpZM4JbutB>
> .
>